api/app/graphql/mutations/auth/sign_in.rb

```diff
@@ -0,0 +1,31 @@
+module Mutations
+    module Auth
+      class SignIn < BaseMutation
+        require 'jwt_service'
+
+  
+        description "Signs In the user into the system"
+  
+        argument :input, Types::Inputs::SignInInput, required: true
+  
+        field :token, String, null: false, description: "User's Authorizations Token to be used in Authenticated mutations and queries"
+        field :user, Types::UserType, null: false, description: "User output"
+  
+        def resolve(input: nil)
+          user = User.find_by(phone: input.phone)
+           if user && user.authenticate(input[:password])
+            token = TokenAuthorization.encode({user_id: user.id})
+            if token
+              return {token: token, user: user}
+            else
+              raise GraphQL::ExecutionError.new("Server was unable to created token. Please try again later.")
+            end
+          else
+            raise GraphQL::ExecutionError.new("Invalid phone number or password.")
+          end
+        end
```

Comment:
Instead of having multiple nested `if` statements, you can use `return` statements. They make the code more readable.
Eg.
```ruby

return error unless user && user.authenticate(...)
token = TokenAuthorization.encode(...)
return error2 unless token
return response


```